### ISSUE TITLE
MaskedConstant should probably set its underlying arrays flags to not-writable

### ISSUE THREAD

**Issue by seberg** [numpy/numpy#9594]
Just as an additional safeguard against attribute usage and array casting, the MaskedConstant should really set its `_data` and `_mask` arrays to be readonly.

**Comment by eric-wieser** [comment]
I thought I did this already?

**Comment by seberg** [comment]
What you have is probably good enough, but thought one could do this just to be sure?
```
In [1]: np.ma.masked.mask[...] = 12

In [2]: np.ma.masked.mask.flags.writeable = False

In [3]: np.ma.masked.mask[...] = 12
ValueError: assignment destination is read-only
````

**Comment by jrbourbeau** [comment]
It looks like (with numpy version `1.13.1`) that `_data` and `_mask` can be modified. For example, 

```python
import numpy as np

a = np.arange(10)
masked_array = np.ma.masked_greater_equal(a, 5)

print('Before modifcation:')
print('masked_array = {}'.format(masked_array))
print('masked_array.data = {}'.format(masked_array.data))
print('masked_array.mask = {}'.format(masked_array.mask))

masked_array.data[3] = -99
masked_array.mask[2] = True
print('After modifcation:')
print('masked_array = {}'.format(masked_array))
print('masked_array.data = {}'.format(masked_array.data))
print('masked_array.mask = {}'.format(masked_array.mask))

```

outputs

```
Before modifcation:
masked_array = [0 1 2 3 4 -- -- -- -- --]
masked_array.data = [0 1 2 3 4 5 6 7 8 9]
masked_array.mask = [False False False False False  True  True  True  True  True]
After modifcation:
masked_array = [0 1 -- -99 4 -- -- -- -- --]
masked_array.data = [  0   1   2 -99   4   5   6   7   8   9]
masked_array.mask = [False False  True False False  True  True  True  True  True]
```


If this behavior isn't desirable, I can submit a PR with the fix @seberg proposed.

**Comment by jrbourbeau** [comment]
Actually, I think I misunderstood the original issue. So my previous comment doesn't apply.

**Comment by eric-wieser** [comment]
I was referring to #9336, which fixed this. Turns out it's not yet merged, waiting on me for release notes.

**Comment by seberg** [comment]
Ah, sorry, didn't realize that was around, never mind then, could probably just close this directly even.

**Comment by eric-wieser** [comment]
Fixed in #9336
